src/ngraph/runtime/gpu/gpu_kernel_emitters.hpp

```diff
@@ -26,6 +28,131 @@ namespace ngraph
+
+                void emit_epilogue(codegen::CodeWriter& writer)
+                {
+                    writer.indent--;
+                    writer << "}\n";
+                }
+
+                void emit_memset(codegen::CodeWriter& writer,
+                                 const GPU_TensorViewWrapper& dst,
+                                 int value,
+                                 size_t buffer_size = 0)
+                {
+                    if (buffer_size == 0)
+                    {
+                        buffer_size = dst.get_size() * dst.get_element_type().size();
+                    }
+                    writer << "runtime::gpu::cuda_memset(" << dst.get_name() << ", " << value
+                           << ", " << buffer_size << ");\n";
+                }
+
+                void emit_memcpyDtD(codegen::CodeWriter& writer,
+                                    const GPU_TensorViewWrapper& dst,
+                                    const GPU_TensorViewWrapper& src)
+                {
+                    writer << "runtime::gpu::cuda_memcpyDtD(" << dst.get_name() << ", "
+                           << src.get_name() << ", " << dst.get_size() << " * "
+                           << dst.get_element_type().size() << ");\n";
+                    return;
+                }
+
+                void emit_cudnnTensor4dDescriptor(codegen::CodeWriter& writer,
+                                                  const std::string& name,
+                                                  const std::string& format,
+                                                  const std::string& data_type,
+                                                  const std::array<size_t, 4>& axes)
+                {
+                    writer << "cudnnTensorDescriptor_t " << name << ";\n";
+                    writer << "cudnnCreateTensorDescriptor(&" << name << ");\n";
+                    writer << "cudnnSetTensor4dDescriptor(" << name << ",\n";
+                    writer << "                 /*format=*/" << format << ",\n";
+                    writer << "                 /*dataType=*/" << data_type;
+                    for (auto const& axis : axes)
+                    {
+                        writer << ",\n                 /*dimension_size*/" << axis;
+                    }
+                    writer << ");\n";
+                }
+
+                void emit_cudnnTensorNdDescriptor(codegen::CodeWriter& writer,
+                                                  const std::string& name,
+                                                  const std::string& data_type,
+                                                  const size_t& num_axes,
+                                                  const std::vector<size_t>& axes,
+                                                  const std::vector<size_t>& strides)
+                {
+                    writer << "const int " << name << "_axes[] = {";
+                    for (auto const& axis : axes)
```

Comment:
Use the util function `join` to do this, and the strides below.

Reply:
Excellent, using this now for the axis and stride emission.